### PR TITLE
cons: make RzLineUndoEntry public in rz_cons.h to fix rz-bindgen

### PR DESCRIPTION
**Pull request by ret2libc**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [ ] I've documented or updated the documentation of every function and struct this PR changes. If not so I've explained why.
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

Although it is not strictly necessary to have `RzLineUndoEntry` in rz_cons.h, it is necessary for rz-bindgen to work correctly because `undo_vec` has a type annotation that references it.

**Test plan**

Build rz-bindgen with this patch. I tested it locally and it works.

**Closing issues**

...

### CODE REVIEW

**Comment by XVilka** [comment]
Strange error on Appveyor clang build:
```
In file included from ../subprojects/xz-5.4.1/src/liblzma/common/common.c:13:
In file included from ..\subprojects\xz-5.4.1\src\liblzma\common/common.h:18:
..\subprojects\xz-5.4.1\src\common\tuklib_integer.h(647,2): error: call to undeclared library function '_BitScanReverse' with type 'unsigned char (unsigned long *, unsigned long)'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
        _BitScanReverse(&i, n);
        ^
..\subprojects\xz-5.4.1\src\common\tuklib_integer.h(647,2): note: include the header <intrin.h> or explicitly provide a declaration for '_BitScanReverse'
..\subprojects\xz-5.4.1\src\common\tuklib_integer.h(749,2): error: call to undeclared library function '_BitScanForward' with type 'unsigned char (unsigned long *, unsigned long)'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
        _BitScanForward(&i, n);
        ^
..\subprojects\xz-5.4.1\src\common\tuklib_integer.h(749,2): note: include the header <intrin.h> or explicitly provide a declaration for '_BitScanForward'
2 errors generated.
[120/1975] Compiling C object subprojects/lz4-1.9.4/liblz4.a.p/lib_lz4.c.obj
[121/1975] Compiling C object subprojects/tree-sitter-0.20.7/libtree_sitter.a.p/lib_src_lib.c.obj
```

Likely because they updated the Clang: `clang-cl (clang-cl 16.0.1)`

File: librz/include/rz_cons.h [hunk 1]
```diff
@@ -1138,6 +1138,20 @@ typedef char *(*RzLineEditorCb)(void *core, const char *str);
...
+ */
+typedef struct rz_line_undo_entry_t {
+	int offset; ///< the beginning index of buffer edit.
+	char *deleted_text; ///< text to be deleted. null-terminated
+	int deleted_len; ///< the length of deleted text
+	char *inserted_text; ///< text to be inserted. null-terminated.
+	int inserted_len; ///< the length of inserted text.
+	bool continuous_next; ///< if true, redo function will continuously process the next entry.
+	bool continuous_prev; ///< if true, undo function will continuously process the previous entry.
+} RzLineUndoEntry;
```
Comment (wargio) [inline librz/include/rz_cons.h]: Why not just make a private header?
```suggestion
typedef struct rz_line_undo_entry_t RzLineUndoEntry;
```

**Review by XVilka — APPROVED** [review]

**Review by wargio — APPROVED** [review]

**Comment by XVilka** [comment]
```
run-time dependency LLVM found: YES 14.0.0
meson.build:35:2: ERROR: Problem encountered: CLang path does not exist
```

**Comment by ret2libc** [comment]
It's green now!

**Review by XVilka — APPROVED** [review]
LGTM

**Pull request merged**